src/biokbase/narrative/tests/test_exception_to_message.py

```diff
@@ -0,0 +1,313 @@
+"""Tests of the ExceptionToMessage class."""
+
+# disable warnings about accessing the JobComm's _comm and _jm attributes
+# ruff: noqa: SLF001
+
+from collections.abc import Callable
+from typing import Any
+
+import pytest
+from biokbase.narrative.exception_util import (
+    JobRequestException,
+    NarrativeException,
+    transform_job_exception,
+)
+from biokbase.narrative.jobs.jobcomm import (
+    MESSAGE_TYPE,
+    ExceptionToMessage,
+    JobComm,
+    JobRequest,
+)
+from biokbase.narrative.tests.job_test_constants import (
+    BATCH_CHILDREN,
+    BATCH_ERROR_RETRIED,
+    BATCH_PARENT,
+    BATCH_TERMINATED,
+    JOB_CREATED,
+    JOB_RUNNING,
+    JOB_TERMINATED,
+)
+from biokbase.narrative.tests.util import make_comm_msg, make_job_request
+
+RQ_TYPE = "request_type"
+CANCEL = MESSAGE_TYPE["CANCEL"]
+ERROR = MESSAGE_TYPE["ERROR"]
+INFO = MESSAGE_TYPE["INFO"]
+STATUS = MESSAGE_TYPE["STATUS"]
+
+NO_JOBS_MAPPING = {
+    "jobs": {},
+    "mapping": {
+        "a": set(),
+        "b": set(),
+        "c": set(),
+    },
+}
+
+"""
+In the code, ExceptionToMessage is called as follows:
+
+    with ExceptionToMessage(msg):
+        request = JobRequest(msg)
+        if request.request_type not in self._msg_map:
+            err_msg = f"Unknown KBaseJobs message '{request.request_type}'"
+            raise JobRequestException(err_msg)
+
+        return self._msg_map[request.request_type](request)
+"""
+
+
+def fn_with_context_manager(
```

Comment:
I think these may have been called `foo` and `bar` or maybe just `fn` -- renamed to be more descriptive and added docs.